File: 程式語言 V2答案檔/C++/503.cpp
#include <iostream>
#include <cmath>
using namespace std;

int main()
{
    int a, b, i;
    double n;

    cin >> a >> b;
    for (i = 1; i < a; i++) {
        n = sqrt(i);
        if ((int)(n) == n) {
            cout << pow(n, b) << endl;
        }
    }
    return (0);
}

File: Programming Language(ver.2)/PHP/503.php
<?php
    $a = intval(fgets(STDIN));
    $b = intval(fgets(STDIN));
    for($i = 1; $i < $a; $i++){
        if(intval(sqrt($i)) != sqrt($i)){
            continue;
        }
        printf("%d\n", pow(sqrt($i), $b));
    }
?>